boms/upper-bounds-check/pom.xml

```diff
@@ -57,11 +57,19 @@
     <dependency>
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-storage</artifactId>
+      <!-- TODO: remove version once new google-cloud-bom with this version comes in -->
```

Comment:
let's discuss at the 4:00 meeting. Perhaps this is happening soon.

Reply:
Yes, it's almost there (PR in java-cloud-bom: https://github.com/googleapis/java-cloud-bom/pull/1874#issuecomment-827777023). But that is blocked by the Libraries BOM (cloud-opensource-java's master branch) having the old google-cloud-core.

Reply:
Did you try switching the order of google-cloud-storage and google-cloud-core in the test pom.xml? In any case, we shouldn't have these Catch-22 problems. I'm not immediately sure how to avoid them, but I think we either have to or reconsider whether this check is worthwhile.